core/trino-main/src/main/java/io/trino/sql/planner/LiteralInterpreter.java

```diff
@@ -125,19 +126,19 @@ protected Object visitDecimalLiteral(DecimalLiteral node, Void context)
         @Override
         protected Slice visitStringLiteral(StringLiteral node, Void context)
         {
-            return node.getSlice();
+            return Slices.utf8Slice(node.getValue());
```

Comment:
AFAIK, StringLiteral contains Slice for performance reasons.
so this is not an OK change.

Reply:
Can we keep byte[] instead there and only wrap as Slice in relevant places?

Reply:
It used to matter when the evaluation engine relied on interpreting the AST for every row. That hasn't been the case for many years.

See previous comment: https://github.com/trinodb/trino/pull/5649#issuecomment-714765359

Reply:
> Can we keep byte[] instead there and only wrap as Slice in relevant places?

This is not needed. The conversions from String in the analyzer/planner don't happen often enough for it to be a meaningful performance issue.

Reply:
Let me benchmark `Slice w/ String` vs `String` vs `byte[] with String view method`

Reply:
I'm benchmarking, so we will be able to compare numbers